lib/as_action_ocaml.ml

```diff
@@ -67,7 +67,7 @@ let compute_deps_ml = compute_deps `Ml
 
 let compile_mli
     ?(needs = []) ?(pkgs = nop) ?(args = []) ~ocamlc ~annot ~incs ~src () =
-  let ctx = As_ctx.v [`OCaml; `Compile; `Src `Mli] in
+  let ctx = As_ctx.v [`OCaml; `Compile; `Src `Mli; `Target `Byte] in
```

Comment:
Why did you add this ? `mli` may be compiled when working on native code generation. In my opinion either `mli` compilation should remain `` `Target`` agnostic or we should distinguish the two (which one could argue should be done because e.g. extension nodes may do something different in the two target, but I don't think it's really feasible with the way the toolchain works).